main.py

```diff
@@ -46,35 +62,56 @@ async def send_chatroommembers_list(websoc):
+    isInvalid = False
     if not websocket in USERS:
         USERS[websocket] = ""
     try:
         async for mesgjson in websocket:
             if sepr in mesgjson and websocket in USERS:
-                if (mesgjson.split(sepr)[0] == "CHKUSR") & (len(mesgjson.split(sepr)) == 3) :
-                    result = str(chk_username_presence(mesgjson))
-                    await websocket.send(result)
-                    if(result == "True"):
-                        await websocket.close()
-                        USERS.pop(websocket)
-                elif USERS[websocket] == "":
-                    USERS[websocket] = [mesgjson.split(sepr)[0], mesgjson.split(sepr)[1]]
-                    print_formatted_text(HTML("[" + obtntime() + "] " + "<b>USERJOINED</b> > <green>" + mesgjson.split(sepr)[0] + "@" + mesgjson.split(sepr)[1] + "</green>"))
-                    await notify_mesej("SNCTRYZERO" + sepr + "USERJOINED" + sepr + mesgjson.split(sepr)[0] + sepr + mesgjson.split(sepr)[1] + sepr + str(getallus(mesgjson.split(sepr)[1])))
+                if USERS[websocket] == "":
+                    if mesgjson.split(sepr)[0] == 'NEW':
+                        new_room = randgene()
+                        current_rooms.append(new_room)
+                        USERS[websocket] = [mesgjson.split(sepr)[1], new_room]
+                        password = Fernet.generate_key().decode("utf-8")
+                        CHATROOM[new_room] = password
+                        await websocket.send(new_room+sepr+password)
+                        print_formatted_text(HTML("[" + obtntime() + "] " + "<b>USERJOINED</b> > <green>" + mesgjson.split(sepr)[1] + "@" + new_room + "</green>"))
+                        await notify_mesej("SNCTRYZERO" + sepr + "USERJOINED" + sepr + mesgjson.split(sepr)[1] + sepr + new_room + sepr + str(getallus(new_room)),new_room)
+                    elif mesgjson.split(sepr)[0] == 'CHKUSR':
+                        # [query username chatroom password]
+                        if mesgjson.split(sepr)[2] in current_rooms and mesgjson.split(sepr)[3] == CHATROOM[mesgjson.split(sepr)[2]] and str(chk_username_presence(mesgjson)) == "False":
+                            await websocket.send("True")
+                            USERS[websocket] = [mesgjson.split(sepr)[1],mesgjson.split(sepr)[2]]
+                            print_formatted_text(HTML("[" + obtntime() + "] " + "<b>USERJOINED</b> > <green>" + mesgjson.split(sepr)[1] + "@" + mesgjson.split(sepr)[2] + "</green>"))
+                            await notify_mesej("SNCTRYZERO" + sepr + "USERJOINED" + sepr + mesgjson.split(sepr)[1] + sepr + mesgjson.split(sepr)[2] + sepr + str(getallus(mesgjson.split(sepr)[2])),mesgjson.split(sepr)[2])
+                        else:
+                            await websocket.send("False")
+                            isInvalid = True
+                    elif str(mesgjson) == '/list':
+                        await send_chatroommembers_list(websocket)
```

Comment:
This is great but try to make an implementation where the message is protected and only the room name is exposed.

Reply:
I am not sure, but using public private key may work. I guess. What do you say?

Reply:
I will work on that and push it.